website/docs/r/virtual_machine.html.markdown

```diff
@@ -134,97 +303,824 @@ routeable network interface within 5 minutes.
+  `datastore_id` in its sub-resource. This also pins it to the specific
+  datastore that is specified - if at a later time the VM and any unpinned
+  disks migrate to another host, the disk will stay on the specified datastore.
+
+An example of datastore pinning is below. As long as the datastore in the
+`pinned_datastore` data source does not change, any change to the standard
+`vm_datastore` data source will not affect the data disk - the disk will stay
+where it is.
+
+```hcl
+resource "vsphere_virtual_machine" "vm" {
+  ...
+
+  datastore_id     = "${data.vsphere_datastore.vm_datastore.id}"
+
+  disk {
+    name = "terraform-test.vmdk"
+    size = 10
+  }
+  
+  disk {
+    datastore_id = "${data.vsphere_datastore.pinned_datastore.id}"
+    name         = "terraform-test_data.vmdk"
+    size         = 100
+    unit_number  = 1
+  }
+
+  ...
+}
+```
+
+## Attribute Reference
+
+The following attributes are exported on the base level of this resource:
+
+* `id` - The UUID of the virtual machine.
+* `reboot_required` - Value internal to Terraform used to determine if a
+  configuration set change requires a reboot. This value is only useful during
+  an update process and gets reset on refresh.
+* `vmware_tools_status` - The state of VMware tools in the guest. This will
+  determine the proper course of action for some device operations.
+* `vmx_path` - The path of the virtual machine's configuration file in the VM's
+  datastore.
+* `imported` - This is flagged if the virtual machine has been imported, or the
+  state has been migrated from a previous version of the resource, and blocks
+  the `clone` configuration option from being set. See the section on
+  [importing](#importing) below.
+* `change_version` - A unique identifier for a given version of the last
+  configuration applied, such the timestamp of the last update to the
+  configuration.
+* `uuid` - The UUID of the virtual machine. Also exposed as the `id` of the
+  resource.
+* `default_ip_address` - The IP address selected by Terraform to be used with
+  any [provisioners][tf-docs-provisioners] configured on this resource.
+  Whenever possible, this is the first IPv4 address that is reachable through
+  the default gateway configured on the machine, then the first reachable IPv6
+  address, and then the first general discovered address if neither exist. If
+  VMware tools is not running on the virtual machine, or if the VM is powered
+  off, this value will be blank.
+* `guest_ip_addresses` - The current list of IP addresses on this machine,
+  including the value of `default_ip_address`. If VMware tools is not running
+  on the virtual machine, or if the VM is powered off, this list will be empty.
+
+## Importing 
+
+An existing virtual machine can be [imported][docs-import] into this resource
+via supplying a JSON string with the full path to the virtual machine, and the
+number of SCSI controllers that the virtual machine has. An example is below:
```

Comment:
can we not determine the number of SCSI controllers automatically so it's possible to just specify a string here?

Reply:
Added this now. The import function now auto-detects this by looking at the first contiguous set of controllers starting at the first bus number - this should catch pretty much all of the cases.